### PR TITLE
Add Missing RP Configs to devconfig

### PR DESCRIPTION
**Pull request by bennerv**
### What this PR does / why we need it:

Missing fields cause `make deploy` to fail because dev-config.yaml target doesn't have all the RP Config fields used

### Test plan for issue:

`make deploy` using this

### Is there any documentation that needs to be updated for this PR?

No

### CODE REVIEW

File: pkg/deploy/devconfig.go [hunk 1]
```diff
@@ -94,7 +94,9 @@ func DevConfig(_env env.Core) (*Config, error) {
 				SubscriptionID:      _env.SubscriptionID(),
 				RPResourceGroupName: os.Getenv("USER") + "-aro-" + _env.Location(),
 				Configuration: &Configuration{
+					AzureCloudName:       &_env.Environment().ActualCloudName,
```
Comment (troy0820) [inline pkg/deploy/devconfig.go:97]:
```suggestion
					AzureCloudName:       to.StrPtr(&_env.Environment().ActualCloudName),
```

File: pkg/deploy/devconfig.go [hunk 2]
```diff
@@ -94,7 +94,9 @@ func DevConfig(_env env.Core) (*Config, error) {
...
 					DatabaseAccountName:  to.StringPtr(os.Getenv("USER") + "-aro-" + _env.Location()),
+					KeyvaultDNSSuffix:    &_env.Environment().KeyVaultDNSSuffix,
```
Comment (troy0820) [inline pkg/deploy/devconfig.go:99]:
```suggestion
					KeyvaultDNSSuffix:    to.StrPtr(&_env.Environment().KeyVaultDNSSuffix),
```

**Review by troy0820 — CHANGES_REQUESTED** [review]
Good catch @bennerv. I believe that these are string pointers as defined in pkg/deploy/config.go.

File: pkg/deploy/devconfig.go (hunk 1 above)
Comment (bennerv) [reply pkg/deploy/devconfig.go:97]: `ActualCloudName` is defined as a string in `pkg/util/azureclient/environments.go`.

File: pkg/deploy/devconfig.go (hunk 2 above)
Comment (bennerv) [reply pkg/deploy/devconfig.go:99]: `KeyVaultDNSSuffix` is a string in azure autorest vendored `environments.go`.

File: pkg/deploy/devconfig.go (hunk 1 above)
Comment (troy0820) [reply pkg/deploy/devconfig.go:97]: using `&` is fine, I'm fine either way. `to.StrPtr` is a little more consistent

File: pkg/deploy/devconfig.go (hunk 2 above)
Comment (troy0820) [reply pkg/deploy/devconfig.go:99]: using `&` is fine, I'm fine either way. `to.StrPtr` is a little more consistent

**Review by jewzaam — APPROVED** [review]

**Pull request merged**